src/main/fc/rc.c

```diff
@@ -265,32 +265,58 @@ static void scaleRawSetpointToFpvCamAngle(void)
     rawSetpoint[YAW]  = constrainf(yaw  * cosFactor + roll * sinFactor, SETPOINT_RATE_LIMIT_MIN, SETPOINT_RATE_LIMIT_MAX);
 }
 
-void updateRcRefreshRate(timeUs_t currentTimeUs)
+void updateRcRefreshRate(timeUs_t currentTimeUs, bool rxReceivingSignal)
 {
-    static timeUs_t lastRxTimeUs;
-
-    timeDelta_t frameAgeUs;
-    timeDelta_t frameDeltaUs = rxGetFrameDelta(&frameAgeUs);
-
-    if (!frameDeltaUs || cmpTimeUs(currentTimeUs, lastRxTimeUs) <= frameAgeUs) {
-        frameDeltaUs = cmpTimeUs(currentTimeUs, lastRxTimeUs);
+    // this function only runs when rxDataProcessingRequired is true, which is:
+    // every time a new frame is detected,
+    // and after RXLOSS_TRIGGER_INTERVAL since the last good frame, and
+    // then every RX_FRAME_RECHECK_INTERVAL, until a new frame is detected
+    // it provides values for use in RCSmoothing, Feedforward, etc.
+    static timeUs_t lastRxTimeUs = 0;
+    timeDelta_t delta = 0;
+
+    if (rxReceivingSignal) { // true while receiving data and until RXLOSS_TRIGGER_INTERVAL expires, otherwise false
```

Comment:
`and until RXLOSS_TRIGGER_INTERVAL expires` - this part should never happen according to description above

Reply:
It is all a bit confusing.  I took some time again to refresh my memory :-)

tl:dr I edited the comments to clarify them, you are correct, the comments were misleading.

I think that in `rxFrameCheck()`, while we are getting Rx data, we allow a wait period of 150ms  aka `RXLOSS_TRIGGER_INTERVAL` before announcing signal loss.  

During this period, we:
- keep looking for new packets, and if one arrives, reset the timers.
- otherwise wait... and stilll say, `rxSignalReceived = true`, which suppresses the `RXLOSS` message
- in rc.c, since the longest allowed reported interval is `RX_INTERVAL_MAX_US`, which is 65.5ms, we report that interval as `currentRxIntervalUs`, even though the true interval may exceed that value.  That's to prevent the filtering and feedforward getting too weird from extremely long loss intervals.
- unless data is received, we do not set rxDataProcessingRequired to true

Once that 150ms period expires, with no packets received, we:
- set `rxSignalReceived = false`, 
- show `RXLOSS`, and 
- start re-checking the Rx for new packets more frequently, at the `RX_FRAME_RECHECK_INTERVAL` of 50ms.
- set `rxDataProcessingRequired` to true, in case failsafe is doing something

We set the `rxIsReceivingSignal()` boolean based on the status of `rxSignalReceived`, and rc.c 

The `updateRcRefreshRate()` function runs from core.c to set the current rate, using the status of `rxSignalReceived `, and that's the code we are looking at above.

I have clarified the comments.